src/types/peer_info.py

```diff
@@ -1,12 +1,59 @@
 from dataclasses import dataclass
 
-from src.util.ints import uint16
+from src.util.ints import uint16, uint64
 from src.util.streamable import Streamable, streamable
 
+import ipaddress
 
-@dataclass(frozen=True)
+
+@dataclass(frozen=True, init=False, eq=False)
 @streamable
 class PeerInfo(Streamable):
     # TODO: Change `host` type to bytes16
     host: str
     port: uint16
+    timestamp: uint64
+
+    def __init__(self, host, port, timestamp=0):
```

Comment:
Why override init and eq?

Reply:
I had in mind creating PeerInfo with no timestamp (so it'll be defaulted to 0). Also, for comparing PeerInfos ignoring the timestamps. I believe the defaults will always require a timestamp and will always use it in comparison as well.

Reply:
I think you might be better off keeping `PeerInfo` as it is, and creating maybe `LastPeerConnection` or `LastConnection` or `PeerTimes` or something which is just `PeerInfo` + `last_connected_timestamp`, especially since `PeerInfo` as a used as a parameter in many places and you don't always want a timestamp to tag along.